### ISSUE TITLE
Alternative Focus on input search 

### ISSUE THREAD

**Issue by geniusupgrader** [infokiller/google-search-navigator#3]
There should be a shortcut to jump to the input search and immediately add some words to the search.
Like it was with google instant, just typing for adding words to the input search. But maybe this is not working with the api?

**Comment by infokiller** [comment]
Would something like `A` (shift+a) work for you here?
(A was chosen because it's similar to the same command in vim).

**Comment by geniusupgrader** [comment]
I'm not a fan of vim. Best Solution is to let the users change the shortcuts.
But do as you wish, you are the maintainer.

**Comment by infokiller** [comment]
Yes, adding customization in the options should be easy, I meant that `A` will be the default.
Just to make sure I understand, you want to be able to type this shortcut and then the search box will get focus and the cursor will be at the end of the current text?

**Comment by rik-smeets** [comment]
@infokiller Feel free to contact me if you're low on time to do this. I can also make the changes if requested. Since I really appreciate you making this extension to begin with, I like to help out further extending or improving it :).

**Comment by infokiller** [comment]
Thanks @rik-smeets !

I'm a bit low on time right now, but I think I'll have time to do a batch of improvements (addressing other issues too) in a few (4-5) weeks.

@geniusupgrader 
Let's wait for more feedback (keep these issues coming!) and I'll try to address most/all of them in the next batch of improvements.

**Comment by Dawa12** [comment]
@infokiller I too love to see this feature added! Love the extension!

**Comment by tedzhu** [comment]
Hi, just want to revive this topic with my two cents. It seems there're two use cases:
1. focus search, select whole thing, to do a new search
2. focus search, move cursor to end, to add more texts
In my own experience I feel 2. is more common. Since making search, refining query is kind of like a session, ending with going into a site.
I'd suggest changing the behavior from 1. to 2. Or, similar to the discussion above, make them two commands with configurable hotkey (default '/' for 1.,  'escape' for 2., maybe?)

Anyway, I'd like to hear what you guys think. I don't mind working on a PR.

**Comment by infokiller** [comment]
I personally rarely run into use case 2. For use case 1, I instinctively open a new tab in chrome and start a search.
So my experience and workflow is probably not that relevant for this feature.

**Comment by tedzhu** [comment]
(+1 for new tab for new search)

Understandable that everyone has different workflow. In that sense I think separating into 2 commands will provide the flexibility.

**Comment by beracira** [comment]
@geniusupgrader I do miss the old google instant search, but the thing is that will be conflicting other shortcuts. One option will be to disable other functionalities so that alphanumeric characters are freed up.

**Comment by ryankhart** [comment]
What I'd really want to have the option for is to set `/` to bring focus to the search bar but *not* select all the text and append instead. If I wanted to replace all the search text with a new one, I'd just open up a new tab and search there.

[Google Search Keyboard Shortcuts](https://chrome.google.com/webstore/detail/google-search-keyboard-sh/iobmefdldoplhmonnnkchglfdeepnfhd) on Chrome works as described above. I recently switched from Chrome to Firefox, and that extension doesn't exist on Firefox at this time.

**Comment by mancze** [comment]
@ryankhart If you activate the "Focus search box" shortcut twice the search box will be focused with cursor at the end of the text. Would it work for you as a workaround?

**Comment by ryankhart** [comment]
> @ryankhart If you activate the "Focus search box" shortcut twice the search box will be focused with cursor at the end of the text. Would it work for you as a workaround?

@mancze Sweet! That's much easier than I thought. I was already trying to build from my own personal fork because I deleted the line of code for selecting. Thanks for showing me that trick.

**Comment by tedzhu** [comment]
> @ryankhart If you activate the "Focus search box" shortcut twice the search box will be focused with cursor at the end of the text. Would it work for you as a workaround?

@ryankhart what do you mean activate it twice? First time it focus the search box, second time the character will be typed in, right?
